src/server/game/Instances/InstanceSaveMgr.cpp

```diff
@@ -286,8 +286,32 @@ void InstanceSaveManager::LoadInstances()
     CharacterDatabase.DirectExecute("UPDATE corpse SET instanceId = 0 WHERE instanceId > 0 AND instanceId NOT IN (SELECT id FROM instance)");
     CharacterDatabase.DirectExecute("UPDATE characters AS tmp LEFT JOIN instance ON tmp.instance_id = instance.id SET tmp.instance_id = 0 WHERE tmp.instance_id > 0 AND instance.id IS NULL");
 
+    // Reduce instance ID numbers when they climb too high, to prevent overflow
+    QueryResult result = CharacterDatabase.Query("SELECT MIN(id),MAX(id) FROM instance");
+    if (result)
+    {
+        uint32 minId = (*result)[0].GetUInt32();
+        uint32 maxId = (*result)[1].GetUInt32();
+
+        // If the gap at the front is larger than the ID range, shift all IDs down to start at 1
+        if (maxId > minId && minId > maxId-minId)
```

Comment:
did you mean "maxId != minId" instead of "maxId > minId" ? the > instead of != looks quite confusing